diff --git a/midl-to-scapy/idl/win/ms-cmrp_clusapi3.idl b/midl-to-scapy/idl/win/ms-cmrp_clusapi3.idl
index 7ab6d1b..0e2eeae 100644
--- a/midl-to-scapy/idl/win/ms-cmrp_clusapi3.idl
+++ b/midl-to-scapy/idl/win/ms-cmrp_clusapi3.idl
@@ -2,10 +2,11 @@
     import "ms-dtyp.idl";
     
+    #define MAX_CLUSTER_CONTROL_CODE_BUFFER_SIZE 0x7FFFFFFF
+
     [
         uuid(b97db8b2-4c63-11cf-bff6-08002be23f2f), 
         version(3.0)
     ]
-    #define MAX_CLUSTER_CONTROL_CODE_BUFFER_SIZE 0x7FFFFFFF
   
     interface clusapi
     {
