diff --git a/midl-to-scapy/idl/win/ms-even6.idl b/midl-to-scapy/idl/win/ms-even6.idl
index fa11ec7..15d5ebe 100644
--- a/midl-to-scapy/idl/win/ms-even6.idl
+++ b/midl-to-scapy/idl/win/ms-even6.idl
@@ -113,7 +113,7 @@
      {
          EvtRpcVariantType type;
          DWORD flags;
-         [switch_is(type)] union
+         [switch_is(type), switch_type(tag_EvtRpcVariantType)] union
          {
              [case(EvtRpcVarTypeNull)] int nullVal;
              [case(EvtRpcVarTypeBoolean)] boolean booleanVal;
