diff --git a/midl-to-scapy/idl/win/ms-eerr.idl b/midl-to-scapy/idl/win/ms-eerr.idl
index b40a5f0..fd1839f 100644
--- a/midl-to-scapy/idl/win/ms-eerr.idl
+++ b/midl-to-scapy/idl/win/ms-eerr.idl
@@ -1,13 +1,13 @@
 // [ms-eerr] v16.0 (Tue, 23 Apr 2024)
  import "ms-dtyp.idl";
   
- [                       
- uuid(14a8831c-bc82-11d2-8a64-0008c7457e5d),
- version(1.0),
- pointer_default(unique)
- ]
- interface ExtendedError
- {
+// [                       
+// uuid(14a8831c-bc82-11d2-8a64-0008c7457e5d),
+// version(1.0),
+// pointer_default(unique)
+// ]
+// interface ExtendedError
+// {
   
  typedef struct tagEEAString
  {
@@ -84,5 +84,5 @@
  } ExtendedErrorInfo;
   
  typedef ExtendedErrorInfo *ExtendedErrorInfoPtr;
- }
+// }
   
