diff --git a/midl-to-scapy/idl/win/ms-fax_fax.idl b/midl-to-scapy/idl/win/ms-fax_fax.idl
index 5b787ae..e674b91 100644
--- a/midl-to-scapy/idl/win/ms-fax_fax.idl
+++ b/midl-to-scapy/idl/win/ms-fax_fax.idl
@@ -1,15 +1,17 @@
  import "ms-fax_faxdatatypes.idl";
- [
-     uuid(ea0a3165-4834-11d2-a6f8-00c04fa346cc),
-     version(4.0),
-     pointer_default(unique)
- ]
-  
+
  #define HCALL DWORD
  #define _FAX_RPC_LIMIT_H
  #define FAX_MAX_RPC_BUFFER                (1024*1024)
  #define FAX_MAX_RECIPIENTS                 10000
  #define RPC_COPY_BUFFER_SIZE   16384  
+
+ [
+     uuid(ea0a3165-4834-11d2-a6f8-00c04fa346cc),
+     version(4.0),
+     pointer_default(unique)
+ ]
+
   
  interface fax
   
