diff --git a/midl-to-scapy/idl/win/ms-mqmr.idl b/midl-to-scapy/idl/win/ms-mqmr.idl
index 7a9d684..63a03b8 100644
--- a/midl-to-scapy/idl/win/ms-mqmr.idl
+++ b/midl-to-scapy/idl/win/ms-mqmr.idl
@@ -19,7 +19,7 @@
   
      typedef struct _MGMT_OBJECT {
          MgmtObjectType type;
-         [switch_is(type)] union
+         [switch_is(type), switch_type(MgmtObjectType)] union
          {
              [case(MGMT_QUEUE)]
                  QUEUE_FORMAT* pQueueFormat;
