diff --git a/midl-to-scapy/idl/win/ms-nrpc.idl b/midl-to-scapy/idl/win/ms-nrpc.idl
index 1cf575e..8a04399 100644
--- a/midl-to-scapy/idl/win/ms-nrpc.idl
+++ b/midl-to-scapy/idl/win/ms-nrpc.idl
@@ -8,6 +8,9 @@
   
  interface logon
  {
+    typedef RPC_UNICODE_STRING UNICODE_STRING;
+    typedef PRPC_UNICODE_STRING PUNICODE_STRING;
+
      typedef struct _STRING{
          USHORT Length;
          USHORT MaximumLength;
@@ -509,6 +512,19 @@
          [size_is(DataLength)] UCHAR * LogonData;
      } NETLOGON_GENERIC_INFO, 
        *PNETLOGON_GENERIC_INFO;
+
+    typedef struct _NETLOGON_TICKET_LOGON_INFO{
+        // Guessed from [MS-APDS] sect 2.2.2.1
+        USHORT CriticalOptions;
+        USHORT ComputerDomainOptions;
+        USHORT TransitOptions;
+        USHORT KerberosOptions;
+        ULONG ServiceTicketLength;
+        [size_is(ServiceTicketLength)] UCHAR * ServiceTicket;
+        ULONG AdditionalTicketLength;
+        [size_is(AdditionalTicketLength)] UCHAR * AdditionalTicket;
+    } NETLOGON_TICKET_LOGON_INFO,
+      *PNETLOGON_TICKET_LOGON_INFO;
   
      typedef [switch_type(NETLOGON_LOGON_INFO_CLASS)] union 
           _NETLOGON_LEVEL{
@@ -539,7 +555,7 @@
          NetlogonValidationSamInfo2 = 3,
          NetlogonValidationGenericInfo = 4,
          NetlogonValidationGenericInfo2 = 5,
-         NetlogonValidationSamInfo4 = 6
+         NetlogonValidationSamInfo4 = 6,
          NetlogonValidationTicketLogon = 7
      } NETLOGON_VALIDATION_INFO_CLASS;
   
@@ -671,6 +687,24 @@
          UNICODE_STRING ExpansionString10;
      } NETLOGON_VALIDATION_SAM_INFO4,
      *PNETLOGON_VALIDATION_SAM_INFO4;
+
+    typedef struct _NETLOGON_VALIDATION_TICKET_LOGON{
+        UCHAR A;
+        UCHAR B;
+        UCHAR C;
+        UCHAR D;
+        USHORT SourceInformation;
+        USHORT TransitInformation;
+        ULONG KerberosStatus;
+        ULONG NetlogonStatus;
+        PNETLOGON_VALIDATION_SAM_INFO4 UserInformation;
+        PNETLOGON_VALIDATION_SAM_INFO4 DeviceInformation;
+        ULONG UserClaimsLength;
+        [size_is(UserClaimsLength)] UCHAR * UserClaims;
+        ULONG DeviceClaimsLength;
+        [size_is(DeviceClaimsLength)] UCHAR * DeviceClaims;
+    } NETLOGON_VALIDATION_TICKET_LOGON,
+    *PNETLOGON_VALIDATION_TICKET_LOGON;
   
      typedef [switch_type(enum _NETLOGON_VALIDATION_INFO_CLASS)] union 
           _NETLOGON_VALIDATION{
