diff --git a/midl-to-scapy/idl/win/ms-ssp.idl b/midl-to-scapy/idl/win/ms-ssp.idl
index e99512e..97cd0c7 100644
--- a/midl-to-scapy/idl/win/ms-ssp.idl
+++ b/midl-to-scapy/idl/win/ms-ssp.idl
@@ -1,4 +1,6 @@
- import "ms-rpce.idl";
+ import "ms-rpce.idl";
+ import "ms-dtyp.idl";
+
  const unsigned long MAX_MASTER_SECRET_BYTES=16;
  typedef [range(MAX_MASTER_SECRET_BYTES,MAX_MASTER_SECRET_BYTES)] unsigned long SizeOfMasterSecretInBytes;
  [
