diff --git a/midl-to-scapy/idl/win/ms-tsts_legacy.idl b/midl-to-scapy/idl/win/ms-tsts_legacy.idl
index c2ce937..a39891b 100644
--- a/midl-to-scapy/idl/win/ms-tsts_legacy.idl
+++ b/midl-to-scapy/idl/win/ms-tsts_legacy.idl
@@ -2,11 +2,11 @@
  import "ms-dtyp.idl";
   
- #include "ms-tsts_allproc.h"
+ import "ms-tsts_allproc.h";
   
  typedef UINT *        UINT_PTR;
   
- static const WCHAR TSRPC_LEGACY_REMOTE_ENDPOINT[] = 
- L"\\pipe\\Ctx_WinStation_API_service";
+//  static const WCHAR TSRPC_LEGACY_REMOTE_ENDPOINT[] = 
+//  L"\\pipe\\Ctx_WinStation_API_service";
   
  typedef [context_handle] void *SERVER_HANDLE;
   
