diff --git a/midl-to-scapy/idl/win/ms-uamg.idl b/midl-to-scapy/idl/win/ms-uamg.idl
index 62e13e3..6cc052a 100644
--- a/midl-to-scapy/idl/win/ms-uamg.idl
+++ b/midl-to-scapy/idl/win/ms-uamg.idl
@@ -155,7 +155,8 @@ typedef  enum tagUpdateServiceOption
 {
      usoNonVolatileService = 0x00000001,
 } UpdateServiceOption;
-typedef DECIMAL DECIMAL;
+
+//typedef DECIMAL DECIMAL;
  
  
 interface ICategoryCollection;
