diff --git a/midl-to-scapy/idl/win/ms-vds.idl b/midl-to-scapy/idl/win/ms-vds.idl
index e51b1c9..00ee6c3 100644
--- a/midl-to-scapy/idl/win/ms-vds.idl
+++ b/midl-to-scapy/idl/win/ms-vds.idl
@@ -808,7 +808,7 @@
  typedef struct VDS_REPARSE_POINT_PROP
  {
      VDS_OBJECT_ID   SourceVolumeId;
-     [string] WCHAR  *pwszPath;
+     [string, max_is(MAX_PATH - 1)] WCHAR  *pwszPath;
  } VDS_REPARSE_POINT_PROP, *PVDS_REPARSE_POINT_PROP;
   
  typedef struct _VDS_DRIVE_LETTER_PROP
